### ISSUE TITLE
choices under a gather increase gather readcount

### ISSUE THREAD

**Issue by facelesspanda** [inkle/ink#539]
the following
```
- (gather)
* choice
- {gather}
```
outputs 2 after choosing the choice.

The json generated looks like this:
```
{
    "inkVersion": 19,
    "root": [
        [
            [
                ["ev", {
                        "^->": "0.gather.0.$r1"
                    }, {
                        "temp=": "$r"
                    }, "str", {
                        "->": ".^.s"
                    },
                    [{
                        "#n": "$r1"
                    }], "/str", "/ev", {
                        "*": ".^.^.c-0",
                        "flg": 18
                    }, {
                        "s": ["^choice", {
                            "->": "$r",
                            "var": true
                        }, null]
                    }
                ], {
                    "c-0": ["ev", {
                            "^->": "0.gather.c-0.$r2"
                        }, "/ev", {
                            "temp=": "$r"
                        }, {
                            "->": ".^.^.0.s"
                        },
                        [{
                            "#n": "$r2"
                        }], "\n", {
                            "->": "0.g-0"
                        }, {
                            "#f": 5
                        }
                    ],
                    "#f": 5,
                    "#n": "gather"
                }
            ], {
                "g-0": ["ev", {
                    "CNT?": "0.gather"
                }, "out", "/ev", "\n", ["done", {
                    "#f": 5,
                    "#n": "g-1"
                }], {
                    "#f": 5
                }]
            }
        ], "done", {
            "#f": 1
        }
    ],
    "listDefs": {}
}
```

The reason why the gather read count goes up is essentially because the choice container's path is `0.gather.0`, so you access `0.gather.0` once at choice-gathering time, and then when choosing the choice, divert to `0.gather.c-0` and from there back to `0.gather.0.s`, so "visiting the first element of the container", so it increments the read count as per the flag "countAtStartOnly".

see how in https://github.com/inkle/ink/blob/91bbf0bcfd32e2cefa8561eb83d581444900dcf0/ink-engine-runtime/Story.cs#L887-L893 the engine checks if we're in the first element when diverting.
Since this is true, readcount goes up.

As for what solution.... I don't know, and I'm out of time, I'm called for dinner.
Hope this helps.

**Comment by facelesspanda** [comment]
maybe change the compilation scheme of labels to have their first element be a `NoOp`?

**Comment by joethephish** [comment]
Really great catch, thanks! And apologies for the months-late reply :)

This is probably exactly what you said, but to reiterate in my own words: The problem was that the `enteringAtStart` variable wasn't taking into account that it might've simply been a container nested at the start of another container, rather than genuinely being the first leaf node in a nested set of containers.

Fix is here: 6bf9988cc3d7fa6c64b3fe30ea588a75db8f6950
